tasks/team-access.yaml

```diff
@@ -56,32 +42,80 @@
     status_code: 200
   register: access_scope_result
 
-# Create all of the roles
+# Create all of the roles, create_role is depend on access_scope_result
 - name: Create Roles
   include_tasks: create_role.yaml
   with_items: "{{ roles }}"
 
-# Now we need to map
+# Retrieve the auth provider ID, auth has been created in CCM so we can just retrive here.
+- name: Get Auth Provider Id
+  uri:
+    # auth_provider.name: OpenShift in vars
+    url: "https://{{ api_endpoint }}/v1/authProviders?name=OpenShift"
+    method: GET
+    validate_certs: no
+    headers:
+      authorization: "{{ authorization }}"
+    body_format: json
+    status_code: 200
+  register: authprovider_result
+
+- name: Keep the auth provider ID for later use
+  set_fact:
+    "auth_provider_openshift_id": "{{ authprovider_result.json.authProviders[0].id }}"
+
+- name: Declare role name that will use cross the script
+  set_fact:
+    "role_name": "app-team-{{ project_set }}"
+
+- name: Create rolebinding if not exists for Azure AD auth provider
+  uri:
+    url: "https://{{ api_endpoint }}/v1/groups"
+    method: GET
+    validate_certs: no
+    headers:
+      authorization: "{{ authorization }}"
+    body_format: json
+    body:
+      roleName: role_name
+    status_code: [200, 500]
+  register: current_group_response
+
+- name: Keep all existing group for later use
+  set_fact:
+    "current_groups": "{{ current_group_response.json.groups }}"
+
+- name: Set current Openshift rules
+  set_fact:
+    "existing_openshift_rules": "{{ current_groups | selectattr('props.authProviderId','equalto',auth_provider_openshift_id) }}"
+
+- name: Filter out updateing project rules because we will recreate all of them in following steps
+  set_fact:
+    "rules_without_updating_group": "{{ existing_openshift_rules | rejectattr('roleName','equalto', role_name) }}"
+
+- name: New User to add
+  set_fact:
+    "new_user": "{{ openshift_username.split(',') }}"
+
+- name: Building new user rule object
+  set_fact:
+    "new_user_rules": "{{ new_user_rules | default([]) + [{'roleName': role_name, 'props':{ 'authProviderId': auth_provider_openshift_id,'key': 'name','value': item}}] }}"
+  with_items: "{{new_user}}"
+
+- name: New user group = group without old project's rules + new project's rules
+  set_fact:
+    new_user_group: "{{ rules_without_updating_group + new_user_rules }}"
```

Comment:
hmmm i don't think this is correct. This will still contain the other rules for `admin` and `None` roles, which might cause problems when other acs playbook runs at the same time.

For the groupsbatch, this would be a better input:
- previous_groups = list of existing rules for this project set specifically
- required_groups = list of expected rules for this project set = list of existing rules for this project set - conflicting users' rules + additional new rules